renku/domain_model/template.py

```diff
@@ -586,3 +588,73 @@ def update(self, template: Template):
         self.metadata["__template_id__"] = template.id
         self.metadata["__automated_update__"] = template.allow_update
         self.immutable_files = template.immutable_files
+
+
+def find_renku_section(lines: List[str]) -> Tuple[int, int]:
+    """Return start and end line numbers of the Renku-specific section."""
+    start = end = -1
+    for index, line in enumerate(lines):
+        if line.startswith("#        Renku-specific section - DO NOT MODIFY        #"):
+            start = index
+        elif line.endswith("#              End Renku-specific section              #"):
+            end = index
+            break
+
+    return start, end
+
+
+def get_renku_section_from_dockerfile(content: str) -> str:
+    """Return the Renku-specific section of the Dockerfile or the whole Dockerfile if it doesn't exist."""
```

Comment:
I think it is better to return None, empty string or raise an error if the renku section does not exist. As a naive user of this function that is the behaviour that I would expect.